README.md

```diff
@@ -40,61 +60,197 @@ import { ReactChatbot } from "@vectara/react-chatbot";
 
 Configure JSX content to render in the messages window when there are no messages to display.
 
-#### `isInitiallyOpen` (optional)
+##### `isInitiallyOpen` (optional)
 
 Set the chat window to be opened on initial render.
 
-#### `inputSize` (optional)
+##### `inputSize` (optional)
+
+Control the size of the input - either "large" (18px) or "medium" (14px)
+
+##### `zIndex` (optional)
+
+Customize the z-index of the chatbot widget
+
+### Use your own views with the useChat hook
+
+Install React-Chatbot:
+
+```shell
+npm install --save @vectara/react-chatbot
+```
+
+Then use the `useChat` hook in your application like this:
+
+```js
+import { useChat } from "@vectara/react-chatbot/lib";
+
+/* snip */
+
+const { sendMessage, messageHistory, isLoading, hasError } = useChat(
+  "CUSTOMER_ID",
+  ["CORPUS_ID_1", "CORPUS_ID_2", "CORPUS_ID_N"],
+  "API_KEY"
+);
+```
+
+The values returned by the hook can be passed on to your custom components as props or used in any way you wish.
+
+#### <u>Hook Values</u>
+
+##### sendMessage: `async ({ query: string; isRetry?: boolean }) => void;`
+
+This is used to send a message to the chat API. Send `true` for the optional `isRetry` flag to if this is retrying a previously failed message. This allows the internal logic to correctly link the next successful answer to the failed query.
 
-Used to control the size of the input - either "large" (18px) or "medium" (14px)
+##### messageHistory: `ChatTurn[]`
+
+This is an array of objects representing question and answer pairs from the entire conversation. Each pair is a `ChatTurn` object, which is typed as:
+
+```
+export type ChatTurn = {
+  // The ID of this turn
+  id: string;
+
+  // The question that the user asked
+  question: string;
+
+  // The answer to the question from the platform
+  answer: string;
+
+  // The results supporting the platform's answer
+  // In usage, you can surface these references to validate the answer given to your user.
+  // See below for the DeserializedSearchResult type definition
+  results: DeserializedSearchResult[];
+}
+
+export type DeserializedSearchResult = {
+  // The ID of this result
+  id: string;
+
+  // Relevant text snippets from the supporting article
+  snippet: {
+    // Text prior to the most relevant content - to help give context
+    pre: string;
+
+    // Relevant content from this article that supports the answer
+    text: string;
+
+    // Text after the most relevant content - to help give context
+    post: string;
+  };
+
+  // The ingestion source - See https://github.com/vectara/vectara-ingest for more details.
+  source: string;
+
+  // The URL of this search result - only defined if this reference was obtained by a webcrawler.
+  url?: string;
+
+  // The title of this reference
+  title?: string;
+
+  // Any other relevant data
+  metadata: Record<string, unknown>;
+};
+```
+
+_For more information on data ingestion, see our [vectara-ingest](https://github.com/vectara/vectara-ingest) repository._
```

Comment:
This seems like a bit of a non-sequitur. We're already linking to Vectara Ingest in a narrower context, above, so maybe we can remove this bit?

Reply:
https://github.com/vectara/react-chatbot/pull/13/commits/908d9f7b3f5ab2e755606cd271302ce1306b7c1e